network/wsPeer.go

```diff
@@ -976,6 +1006,12 @@ func (wp *wsPeer) makeResponseChannel(key uint64) (responseChannel chan *Respons
 	return newChan
 }
 
+func (wp *wsPeer) hasOutstandingRequests() bool {
+	wp.responseChannelsMutex.Lock()
+	defer wp.responseChannelsMutex.Unlock()
+	return len(wp.responseChannels) > 0
+}
+
```

Comment:
Since you're doing all the work of taking the lock, could you isntead return the `len` directly, and let the caller decide to compare it with 0?

Reply:
I think I have a slight preference for the way it is currently but happy to change if there's a +1 . 

I just don't think that we will be checking the length of this outside of this use-case and to me this parses slightly easier in the conditional.